cli/src/wallet_cli/mod.rs

```diff
@@ -986,32 +987,36 @@ async fn print_wallet_address(wallet: &Wallet) -> Result<(), Error> {
         output_ids.push(output_id);
 
         // Output might be associated with the address, but can't be unlocked by it, so we check that here.
-        let (required_address, _) = &output_data
-            .output
-            .required_and_unlocked_address(slot_index, &output_id)?;
-
-        if address.inner() == required_address {
-            if let Some(nt) = output_data.output.native_token() {
-                native_tokens.add_native_token(*nt)?;
-            }
-            match &output_data.output {
-                Output::Basic(_) => {}
-                Output::Account(account) => accounts.push(account.account_id_non_null(&output_id)),
-                Output::Foundry(foundry) => foundries.push(foundry.id()),
-                Output::Nft(nft) => nfts.push(nft.nft_id_non_null(&output_id)),
-                Output::Delegation(delegation) => delegations.push(delegation.delegation_id_non_null(&output_id)),
-                Output::Anchor(anchor) => anchors.push(anchor.anchor_id_non_null(&output_id)),
+        let required_address = &output_data.output.required_address(
+            slot_index,
+            protocol_parameters.min_committable_age(),
+            protocol_parameters.max_committable_age(),
+        )?;
+
+        if let Some(required_address) = required_address {
+            if address.inner() == required_address {
```

Comment:
I prefer using `matches!` macro here to get rid of the nested ifs.

Reply:
actually better use `if_some_and`

Reply:
What's `if_some_and`?

Reply:
I don't think that's stable yet

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some_and

stable since 1.70

Reply:
sorry ... typo ... `is_some_and`

Reply:
Changed